autogluon/utils/tabular/ml/models/lgb/hyperparameters/lgb_trial.py

```diff
@@ -2,47 +2,62 @@
         import lightgbm as lgb
-        dataset_train = lgb.Dataset(args.directory + args.dataset_train_filename)
-        dataset_val = lgb.Dataset(args.directory + args.dataset_val_filename)
+        task_id = args.pop('task_id')
+        util_args = args.pop('util_args')
+        dataset_train = lgb.Dataset(util_args.directory + util_args.dataset_train_filename)
+        dataset_val = lgb.Dataset(util_args.directory + util_args.dataset_val_filename)
+        X_val, Y_val = load_pkl.load(util_args.directory + util_args.dataset_val_pkl_filename)
 
-        nonparam_args = {'directory', 'task_id', 'time_start', 'time_limit', 'model', 'dataset_train_filename', 'dataset_val_filename'}  # list of args which are not model hyperparameters
         # Note: args.task_id may not start at 0 if HPO has been run for other models with same scheduler
-        file_prefix = "trial_" + str(args.task_id)  # append to all file names created during this trial. Do NOT change!
-        model = args.model
+        file_prefix = "trial_" + str(task_id)  # append to all file names created during this trial. Do NOT change!
+
+        model = util_args.model
         model.name = model.name + os.path.sep + file_prefix
         model.set_contexts(path_context=model.path_root + model.name + os.path.sep)
-        for key in args:
-            if key not in nonparam_args:
-                model.params[key] = args[key]  # use these hyperparam values in this trial
 
         time_current = time.time()
-        time_elapsed = time_current - args.time_start
-        if 'time_limit' in args:
-            time_left = args.time_limit - time_elapsed
+        time_elapsed = time_current - util_args.time_start
+        if 'time_limit' in util_args:
+            time_left = util_args.time_limit - time_elapsed
             if time_left <= 0:
                 raise TimeLimitExceeded
         else:
             time_left = None
 
+        model.params.update(args)
+        time_fit_start = time.time()
         model.fit(dataset_train=dataset_train, dataset_val=dataset_val, time_limit=time_left, reporter=reporter)
-        # TODO: Set fit_time variable in model here
+        time_fit_end = time.time()
+        score = model.score(X=X_val, y=Y_val)
+        time_pred_end = time.time()
+        model.fit_time = time_fit_end - time_fit_start
+        model.predict_time = time_pred_end - time_fit_end
+        model.val_score = score
         model.save()
-        # TODO: add to reporter: time_of_trial without load/save time (isn't this just function of early-stopping point?), memory/inference ??
-        # TODO: add to reporter: Name/path of model
     except Exception as e:
         if not isinstance(e, TimeLimitExceeded):
             logger.exception(e, exc_info=True)
         reporter.terminate()
+
+    # FIXME: If stopping metric and eval metric differ, the previous reported scores will not align as they will be evaluated with stopping_metric, whereas this is evaluated with eval_metric
```

Comment:
Will this also affect the search if it is adaptive?  Ie. when searcher selects next configuration based on the previous configuration's reported score.

If so, we can probably just manually overwrite the final reporter-score with a final `model.score()` call whenever `eval_metric != stopping_metric`.

Reply:
Yes it will, but manually overwriting might not be the best way to do things. If you stick with the same metric across the entire HPO it won't matter (but it will optimize for stopping metric instead of eval metric, which is kind of the intent anyways)